### PR TITLE
feat: add support to archive repositories

### PR DESCRIPTION
**Pull request by dirien**
This PR add support to archive a provider.

👀 @ringods @tmeckel

### CODE REVIEW

**Comment by dirien** [comment]
Not sure why both jobs are failing. 🤔

**Comment by tmeckel** [comment]
> Not sure why both jobs are failing. 🤔

@dirien it's because you want to merge a branch from a fork. The repo is still not configured in a way that secrets are passed to workflow runs that started from a fork. This is an outstanding issue that @ringods and I worked upon but never came to a satisfiable solution.

**Comment by dirien** [comment]
@tmeckel: I have no permission to create a branch in this repo.

**Comment by tmeckel** [comment]
@dirien no need to close the PR. I'll review and merge when I've time. Sorry if my initial answer was misunderstandable in that way.

**Comment by dirien** [comment]
@tmeckel sounds good! Thanks!

File: src/github/repositories.ts [hunk 1]
```diff
@@ -211,6 +214,7 @@ export function configureRepositories(repositoryArgs: Repository[], allTeams: Ma
                     import: repositoryInfo.import || false,
                     template: repositoryInfo.template,
                     removable: repositoryInfo.removable || false,
+                    archived: repositoryInfo.archived || false,
```
Comment (tmeckel) [inline src/github/repositories.ts:217]: Should be `archived: repository.archived === true ? true : false,`

File: src/github/repositories.ts [hunk 2]
```diff
@@ -197,6 +199,7 @@ export function configureRepositories(repositoryArgs: Repository[], allTeams: Ma
...
```
Comment (tmeckel) [inline src/github/repositories.ts:202]: Should be archived: `repository.archived === true ? true : false,`

File: src/github/repositories.ts [hunk 3]
```diff
@@ -42,7 +43,8 @@ abstract class BaseRepository extends pulumi.ComponentResource {
                 template: args.template ? {
                     owner: 'pulumi',
                     repository: args.template,
-                } : undefined
+                } : undefined,
+                archived: args.archived || false
```
Comment (tmeckel) [inline src/github/repositories.ts:47]: Should be archived: `args.archived === true ? true : false,`

**Review by tmeckel — CHANGES_REQUESTED** [review]
I would set `hasIssues` to `false`, if a repository is archived `hasIssues:  args.archived === true ? false : true`. Because it makes no sense to enable issues on an archived repository.

Same with `vulnerabilityAlerts`: `vulnerabilityAlerts: args.archived === true ? false : true,`

**Comment by dirien** [comment]
@tmeckel done

**Comment by tmeckel** [comment]
Approved, after running `pulumi preview` locally with the following result

image

**Pull request merged**